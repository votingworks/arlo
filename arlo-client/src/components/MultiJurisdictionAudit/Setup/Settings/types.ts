import { IAuditSettings } from '../../../../types'

export type IValues = Pick<
  IAuditSettings,
  'electionName' | 'online' | 'randomSeed' | 'riskLimit'
>
