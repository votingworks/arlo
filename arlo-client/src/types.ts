export type ElementType<
  T extends readonly unknown[]
> = T extends readonly (infer ElementType)[] ? ElementType : never

export interface IErrorResponse {
  errors: {
    message: string
  }[]
}

export interface ICreateAuditParams {
  electionId: string
}

export interface IAuditFlowParams extends ICreateAuditParams {
  token: string
  roundId?: string
  ballotId?: string
}

export interface ICandidate {
  id: string
  name: string
  numVotes: number | string
}

export interface ISampleSizeOption {
  size: number | string
  prob: number | null
  type: string | null
}

export interface IContest {
  id: string
  isTargeted: boolean
  name: string
  numWinners: string
  votesAllowed: string
  choices: ICandidate[]
  totalBallotsCast: string
}

export interface IBallot {
  timesSampled: number
  auditBoard?: Pick<IAuditBoard, 'id' | 'name'>
  status: 'AUDITED' | null
  vote: string
  comment: string
  position: number
  batch: {
    id: string
    name: string
    tabulator: string | null
  }
}

export interface IAuditBoardMember {
  name: string
  affiliation: string
}

export interface IReview {
  vote: IBallot['vote']
  comment: IBallot['comment']
}

export interface IAuditBoard {
  id: string
  name: string
  members: IAuditBoardMember[]
  passphrase?: string
  ballots?: IBallot[] // TODO remove
}

export interface IBallotManifest {
  filename: string | null
  numBallots: number | '' | null
  numBatches: number | '' | null
  uploadedAt: string | null
}

export interface IBatch {
  id: string
  name: string
  numBallots: number
  storageLocation: null | string
  tabulator: null | string
}

export interface IJurisdiction {
  id: string
  name: string
  contests: string[]
  auditBoards: IAuditBoard[]
  ballotManifest?: IBallotManifest
  batches?: IBatch[] // optional until I'm ready to update everything
}

export interface IRoundContest {
  id: string
  results: {
    [key: string]: number
  }
  sampleSizeOptions: ISampleSizeOption[] | null
  sampleSize: number | null
  endMeasurements: {
    isComplete: null | boolean
    pvalue: null | number
  }
}

export interface IRound {
  id: string
  name?: string
  randomSeed?: string
  riskLimit?: number
  contests: IRoundContest[]
  startedAt: string
  endedAt: string | null
  jurisdictions?: {
    [key: string]: {
      numBallots: number
    }
  }
}

export interface IAudit {
  name: string
  online: boolean
  riskLimit: string
  randomSeed: string
  contests: IContest[]
  jurisdictions: IJurisdiction[]
  rounds: IRound[]
  frozenAt: string | null
}

export interface IElectionMeta {
  id: string
  name: string
  state: string
  electionDate: string
}

export interface IOrganizationMeta {
  id: string
  name: string
  elections: IElectionMeta[]
}

export interface IJurisdictionMeta {
  id: string
  name: string
  election: IElectionMeta
}

export interface IUserMeta {
  name: string
  email: string
  type: 'audit_admin' | 'jurisdiction_admin'
  organizations: IOrganizationMeta[]
  jurisdictions: IJurisdictionMeta[]
}

export interface IAuthData {
  isAuthenticated: boolean
  meta?: IUserMeta
}
