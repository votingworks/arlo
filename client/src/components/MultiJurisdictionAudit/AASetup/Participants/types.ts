import { IAuditSettings } from '../../../../types'

export interface IValues {
  state: IAuditSettings['state']
}
