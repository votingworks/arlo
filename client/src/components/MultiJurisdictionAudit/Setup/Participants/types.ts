import { IAuditSettings } from '../../../../types'

export interface IValues {
  csv: File | null
  state: IAuditSettings['state']
}
