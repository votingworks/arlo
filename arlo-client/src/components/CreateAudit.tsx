import React, { useState } from 'react'
import styled from 'styled-components'
import { RouteComponentProps } from 'react-router-dom'
import FormButton from './Form/FormButton'
import api from './utilities'
import { Params } from '../types'

const Button = styled(FormButton)`
  margin: 65px 0;
`

const Wrapper = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 400px;
  text-align: center;
`

const CreateAudit = ({ history }: RouteComponentProps<Params>) => {
  const [loading, setLoading] = useState(false)
  const onClick = async () => {
    setLoading(true)
    const { electionId } = await api('/election/new', {
      electionId: '',
      method: 'POST',
    })
    history.push(`/election/${electionId}`)
  }
  return (
    <Wrapper>
      <img height="50px" src="/arlo.png" alt="Arlo, by VotingWorks" />
      <Button
        type="button"
        intent="primary"
        fill
        large
        onClick={onClick}
        loading={loading}
        disabled={loading}
      >
        Create a New Audit
      </Button>
    </Wrapper>
  )
}

export default CreateAudit
