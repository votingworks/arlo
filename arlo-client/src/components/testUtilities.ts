import { createLocation, createMemoryHistory } from 'history'
import { match as routerMatch } from 'react-router-dom'

type MatchParameter<Params> = { [K in keyof Params]?: string }

const generateUrl = <Params extends MatchParameter<Params>>(
  path: string,
  params: Params
): string => {
  let tempPath = path

  for (const param in params) {
    /* istanbul ignore else */
    if (params.hasOwnProperty(param)) {
      const value = params[param]
      tempPath = tempPath.replace(`:${param}`, value as NonNullable<
        typeof value
      >)
    }
  }

  return tempPath
}

export const routerTestProps = <Params extends MatchParameter<Params> = {}>(
  path: string,
  params: Params
) => {
  const match: routerMatch<Params> = {
    isExact: false,
    path,
    url: generateUrl(path, params),
    params,
  }
  const history = createMemoryHistory()
  const location = createLocation(match.url)

  return { history, location, match }
}

/** Credit to https://stackoverflow.com/a/56452779 for solution to mocking React Router props */

export const asyncForEach = async <T>(
  array: T[],
  callback: (value: T, index: number, array: T[]) => Promise<void>
) => {
  for (let index = 0; index < array.length; index++) {
    await callback(array[index], index, array)
  }
}

export default {
  asyncForEach,
  routerTestProps,
}
