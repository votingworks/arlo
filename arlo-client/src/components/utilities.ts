import { Params } from '../types'

export const api = <T>(
  endpoint: string,
  { electionId, ...options }: Params & RequestInit
): Promise<T> => {
  const apiBaseURL = electionId ? `/election/${electionId}` : ''
  return fetch(apiBaseURL + endpoint, options).then(res => {
    if (!res.ok) {
      throw new Error(res.statusText)
    }
    return res.json() as Promise<T>
  })
}

export default api
