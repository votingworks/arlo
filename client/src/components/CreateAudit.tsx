import React, { useState } from 'react'
import styled from 'styled-components'
import { RouteComponentProps, Link } from 'react-router-dom'
import { RadioGroup, Radio } from '@blueprintjs/core'
import { Formik, FormikProps, Field } from 'formik'
import FormButton from './Atoms/Form/FormButton'
import { api } from './utilities'
import { ICreateAuditParams, IAuditSettings } from '../types'
import { useAuthDataContext } from './UserContext'
import FormSection from './Atoms/Form/FormSection'
import FormField from './Atoms/Form/FormField'

const Button = styled(FormButton)`
  margin: 30px 0 0 0;
`

const Wrapper = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  text-align: center;
  @media (min-width: 500px) {
    width: 400px;
  }
`

const AuditLink = styled(Link)`
  display: block;
  margin: 10px;

  &:first-of-type {
    margin-top: 30px;
  }
`

const CenterField = styled(FormField)`
  width: 100%;
`

const LeftRadioGroup = styled(RadioGroup)`
  padding: 20px;
  text-align: left;
`

interface IValues {
  auditName: string
  auditType: IAuditSettings['auditType']
}

const CreateAudit = ({ history }: RouteComponentProps<ICreateAuditParams>) => {
  const { isAuthenticated, meta } = useAuthDataContext()

  const [loading, setLoading] = useState(false)
  const onSubmit = async ({ auditName, auditType }: IValues) => {
    setLoading(true)
    const response = await api<{ electionId: string }>('/election/new', {
      method: 'POST',
      body: JSON.stringify({
        organizationId: meta!.organizations[0].id,
        auditName,
        auditType,
        isMultiJurisdiction: true,
      }),
      headers: {
        'Content-Type': 'application/json',
      },
    })
    if (!response) {
      setLoading(false)
      return
    }
    const { electionId } = response
    history.push(`/election/${electionId}/setup`)
  }

  if (isAuthenticated === null) return null // Still loading
  return (
    <Wrapper>
      <img height="50px" src="/arlo.png" alt="Arlo, by VotingWorks" />
      {isAuthenticated && meta!.type === 'audit_admin' && (
        <Formik
          onSubmit={onSubmit}
          initialValues={{ auditName: '', auditType: 'BALLOT_POLLING' }}
        >
          {({ handleSubmit, setFieldValue, values }: FormikProps<IValues>) => (
            <>
              <FormSection>
                {/* eslint-disable jsx-a11y/label-has-associated-control */}
                <label htmlFor="audit-name" id="audit-name-label">
                  Give your new audit a unique name.
                  <Field
                    id="audit-name"
                    aria-labelledby="audit-name-label"
                    name="auditName"
                    disabled={loading}
                    validate={(v: string) => (v ? undefined : 'Required')}
                    component={CenterField}
                  />
                </label>
              </FormSection>
              <FormSection>
                <label htmlFor="auditType">
                  Audit type:
                  <LeftRadioGroup
                    name="auditType"
                    onChange={e =>
                      setFieldValue('auditType', e.currentTarget.value)
                    }
                    selectedValue={values.auditType}
                  >
                    <Radio value="BALLOT_POLLING">Ballot Polling</Radio>
                    <Radio value="BATCH_COMPARISON">Batch Comparison</Radio>
                  </LeftRadioGroup>
                </label>
              </FormSection>
              <Button
                type="button"
                intent="primary"
                fill
                large
                onClick={handleSubmit}
                loading={loading}
                disabled={loading}
              >
                Create a New Audit
              </Button>
            </>
          )}
        </Formik>
      )}
      {!isAuthenticated || (meta && meta.type === 'audit_board') ? (
        <>
          <Button
            verticalSpaced
            type="button"
            intent="primary"
            fill
            large
            onClick={
              /* istanbul ignore next */
              () => window.location.replace('/auth/auditadmin/start')
            }
            loading={loading}
            disabled={loading}
          >
            Log in as an Audit Admin
          </Button>
          <Button
            verticalSpaced
            type="button"
            intent="primary"
            fill
            large
            onClick={
              /* istanbul ignore next */
              () => window.location.replace('/auth/jurisdictionadmin/start')
            }
            loading={loading}
            disabled={loading}
          >
            Log in as a Jurisdiction Admin
          </Button>
        </>
      ) : (
        <>
          {meta!.organizations.length > 0 &&
            meta!.organizations.map(o =>
              o.elections.map(election => (
                <AuditLink
                  to={`/election/${election.id}`}
                  key={election.id}
                  className="bp3-button bp3-intent-primary"
                >
                  {election.auditName}
                  {election.state && ` (${election.state})`}
                </AuditLink>
              ))
            )}
          {meta!.jurisdictions.length > 0 &&
            meta!.jurisdictions.map(({ id, election }) => (
              <AuditLink
                to={`/election/${election.id}/jurisdiction/${id}`}
                key={election.id}
                className="bp3-button bp3-intent-primary"
              >
                {election.auditName}
                {election.state && ` (${election.state})`}
              </AuditLink>
            ))}
        </>
      )}
    </Wrapper>
  )
}

export default CreateAudit
