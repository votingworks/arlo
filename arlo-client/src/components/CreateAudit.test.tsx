import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { toast } from 'react-toastify'
import { RouteComponentProps, BrowserRouter as Router } from 'react-router-dom'
import CreateAudit from './CreateAudit'
import { ICreateAuditParams } from '../types'
import { routerTestProps } from './testUtilities'
import * as utilities from './utilities'
import AuthDataProvider from './UserContext'

const apiMock: jest.SpyInstance<
  ReturnType<typeof utilities.api>,
  Parameters<typeof utilities.api>
> = jest.spyOn(utilities, 'api').mockImplementation()
const checkAndToastMock: jest.SpyInstance<
  ReturnType<typeof utilities.checkAndToast>,
  Parameters<typeof utilities.checkAndToast>
> = jest.spyOn(utilities, 'checkAndToast').mockReturnValue(false)

checkAndToastMock.mockReturnValue(false)

const routeProps: RouteComponentProps<ICreateAuditParams> = routerTestProps(
  '/audit/:electionId',
  {
    electionId: '1',
  }
)

const toastSpy = jest.spyOn(toast, 'error').mockImplementation()

const historySpy = jest.spyOn(routeProps.history, 'push').mockImplementation()

afterEach(() => {
  apiMock.mockClear()
  checkAndToastMock.mockClear()
  toastSpy.mockClear()
  historySpy.mockClear()
})

describe('CreateAudit', () => {
  it('renders correctly', () => {
    const { container } = render(<CreateAudit {...routeProps} />)
    expect(container).toMatchSnapshot()
  })

  it('calls the /election/new endpoint for nonauthenticated user', async () => {
    apiMock.mockImplementation(async () => ({ electionId: '1' }))
    const { getByText, getByLabelText } = render(
      <CreateAudit {...routeProps} />
    )

    fireEvent.change(getByLabelText('Give your new audit a unique name.'), {
      target: { value: 'Audit Name' },
    })

    fireEvent.click(getByText('Create a New Audit'), { bubbles: true })

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(apiMock).toHaveBeenNthCalledWith(1, '/election/new', {
        method: 'POST',
        body: JSON.stringify({
          auditName: 'Audit Name',
          isMultiJurisdiction: false,
        }),
        headers: {
          'Content-Type': 'application/json',
        },
      })
      expect(historySpy).toBeCalledTimes(1)
      expect(historySpy).toHaveBeenNthCalledWith(1, '/audit/1/setup')
    })
  })

  it('requires an audit name', async () => {
    apiMock.mockImplementation(async () => ({ electionId: '1' }))
    const { getByText } = render(<CreateAudit {...routeProps} />)

    fireEvent.click(getByText('Create a New Audit'), { bubbles: true })

    await wait(() => {
      expect(getByText('Required')).toBeTruthy()
      expect(apiMock).toBeCalledTimes(0)
    })
  })

  it('calls the /election/new endpoint for authenticated user', async () => {
    apiMock
      .mockImplementationOnce(async () => ({
        type: 'audit_admin',
        name: 'Joe',
        email: 'test@email.org',
        jurisdictions: [],
        organizations: [
          {
            id: 'org-id',
            name: 'State',
            elections: [],
          },
        ],
      }))
      .mockImplementationOnce(async () => ({ electionId: '1' }))
    const { getByText, getByLabelText } = render(
      <AuthDataProvider>
        <CreateAudit {...routeProps} />
      </AuthDataProvider>
    )

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(apiMock).toHaveBeenNthCalledWith(1, '/auth/me')
    })
    fireEvent.change(getByLabelText('Give your new audit a unique name.'), {
      target: { value: 'Audit Name' },
    })
    fireEvent.click(getByText('Create a New Audit'), { bubbles: true })

    await wait(() => {
      expect(apiMock).toBeCalledTimes(2)
      expect(apiMock).toHaveBeenNthCalledWith(2, '/election/new', {
        method: 'POST',
        body: JSON.stringify({
          organizationId: 'org-id',
          auditName: 'Audit Name',
          isMultiJurisdiction: true,
        }),
        headers: {
          'Content-Type': 'application/json',
        },
      })
      expect(historySpy).toBeCalledTimes(1)
      expect(historySpy).toHaveBeenNthCalledWith(1, '/election/1/setup')
    })
  })

  it('lists associated elections for authenticated AA user', async () => {
    apiMock.mockImplementationOnce(async () => ({
      type: 'audit_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [],
      organizations: [
        {
          id: 'org-id',
          name: 'State',
          elections: [
            {
              id: 'election-1',
              auditName: '',
              state: 'NY',
            },
            {
              id: 'election-2',
              auditName: 'Election Two',
              state: 'FL',
            },
            {
              id: 'election-3',
              auditName: 'Election Three',
              state: '',
            },
            {
              id: 'election-4',
              auditName: 'Election Four',
              state: 'WA',
            },
          ],
        },
      ],
    }))
    const { container } = render(
      <Router>
        <AuthDataProvider>
          <CreateAudit {...routeProps} />
        </AuthDataProvider>
      </Router>
    )

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(apiMock).toHaveBeenNthCalledWith(1, '/auth/me')
      expect(container).toMatchSnapshot()
    })
  })

  it('lists associated elections for authenticated JA user', async () => {
    apiMock.mockImplementationOnce(async () => ({
      type: 'audit_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [
        {
          id: 'jurisdiction-1',
          name: 'County One',
          election: {
            id: 'election-1',
            auditName: 'Election One',
            state: 'NY',
          },
        },
        {
          id: 'jurisdiction-2',
          name: 'County Two',
          election: {
            id: 'election-2',
            auditName: '',
            state: '',
          },
        },
      ],
      organizations: [],
    }))
    const { container } = render(
      <Router>
        <AuthDataProvider>
          <CreateAudit {...routeProps} />
        </AuthDataProvider>
      </Router>
    )

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(apiMock).toHaveBeenNthCalledWith(1, '/auth/me')
      expect(container).toMatchSnapshot()
    })
  })

  it('handles error responses from server', async () => {
    apiMock.mockImplementation(async () => ({ electionId: '1' }))
    checkAndToastMock.mockReturnValue(true)
    const { getByText, getByLabelText } = render(
      <CreateAudit {...routeProps} />
    )

    fireEvent.change(getByLabelText('Give your new audit a unique name.'), {
      target: { value: 'Audit Name' },
    })
    fireEvent.click(getByText('Create a New Audit'), { bubbles: true })

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(apiMock.mock.calls[0][0]).toBe('/election/new')
      expect(checkAndToastMock).toBeCalledTimes(1)
      expect(historySpy).toBeCalledTimes(0)
      expect(toastSpy).toBeCalledTimes(0)
    })
  })

  it('handles 404 responses from server', async () => {
    apiMock.mockImplementation(async () => {
      throw new Error('404')
    })
    checkAndToastMock.mockReturnValue(true)
    const { getByText, getByLabelText } = render(
      <CreateAudit {...routeProps} />
    )

    fireEvent.change(getByLabelText('Give your new audit a unique name.'), {
      target: { value: 'Audit Name' },
    })
    fireEvent.click(getByText('Create a New Audit'), { bubbles: true })

    await wait(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(apiMock.mock.calls[0][0]).toBe('/election/new')
      expect(checkAndToastMock).toBeCalledTimes(0)
      expect(historySpy).toBeCalledTimes(0)
      expect(toastSpy).toBeCalledTimes(1)
    })
  })
})
