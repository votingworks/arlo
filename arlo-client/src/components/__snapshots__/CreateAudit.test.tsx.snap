// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAudit lists associated elections for authenticated AA user 1`] = `
<div>
  <div
    class="sc-gZMcBi coLhGN"
  >
    <img
      alt="Arlo, by VotingWorks"
      height="50px"
      src="/arlo.png"
    />
    <div
      class="sc-htpNat kLGmUl"
    >
      <label
        for="audit-name"
        id="audit-name-label"
      >
        Give your new audit a unique name.
        <div
          class="sc-VigVT kxLHnZ sc-EHOje bXavad"
        >
          <div
            class="bp3-input-group sc-bZQynM fFGKEg"
          >
            <input
              aria-labelledby="audit-name-label"
              class="bp3-input"
              id="audit-name"
              name="auditName"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <button
      class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-iwsKbI kljBkO sc-bwzfXH eBfcXh"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Create a New Audit
      </span>
    </button>
  </div>
</div>
`;

exports[`CreateAudit lists associated elections for authenticated JA user 1`] = `
<div>
  <div
    class="sc-gZMcBi coLhGN"
  >
    <img
      alt="Arlo, by VotingWorks"
      height="50px"
      src="/arlo.png"
    />
    <div
      class="sc-htpNat kLGmUl"
    >
      <label
        for="audit-name"
        id="audit-name-label"
      >
        Give your new audit a unique name.
        <div
          class="sc-VigVT kxLHnZ sc-EHOje bXavad"
        >
          <div
            class="bp3-input-group sc-bZQynM fFGKEg"
          >
            <input
              aria-labelledby="audit-name-label"
              class="bp3-input"
              id="audit-name"
              name="auditName"
              style="padding-right: 10px;"
              value=""
            />
          </div>
        </div>
      </label>
    </div>
    <button
      class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-iwsKbI kljBkO sc-bwzfXH eBfcXh"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Create a New Audit
      </span>
    </button>
  </div>
</div>
`;

exports[`CreateAudit renders correctly 1`] = `
<div>
  <div
    class="sc-gZMcBi coLhGN"
  >
    <img
      alt="Arlo, by VotingWorks"
      height="50px"
      src="/arlo.png"
    />
    <button
      class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-iwsKbI kljBkO sc-bdVaJa gLVjYw"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Log in as an Audit Admin
      </span>
    </button>
    <button
      class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-iwsKbI kljBkO sc-bdVaJa gLVjYw"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Log in as a Jurisdiction Admin
      </span>
    </button>
  </div>
</div>
`;
